invoices/invoiceregistry.go

```diff
@@ -261,6 +261,11 @@ func (i *InvoiceRegistry) Stop() error {
 	log.Info("InvoiceRegistry shutting down...")
 	defer log.Debug("InvoiceRegistry shutdown complete")
 
+	if i.expiryWatcher == nil {
+		return fmt.Errorf("InvoiceRegistry expiryWatcher not " +
+			"initialized")
+	}
+
```

Comment:
i think we still want the rest of the function to run though. iiuc, the whole reason we want to call Stop before Start so that quit channels can be closed & hence synch processes in Start methods can be stopped

Reply:
agree changed it to:

```
	if i.expiryWatcher != nil {
		i.expiryWatcher.Stop()
	}
```

Reply:
we could also change the constructor of the invoice registry to catch the case where we have a `nil` pointer for the expiryWatcher. Went with the above for now, but happy to change it.